test/base-tlsutility.cpp

```diff
@@ -85,71 +46,230 @@ BOOST_AUTO_TEST_CASE(sha1)
 	}
 }
 
-BOOST_AUTO_TEST_CASE(iscauptodate_ok)
+static String GetOpenSSLError()
 {
-	auto key (GenKeypair());
-
-	BOOST_CHECK(IsCaUptodate(MakeCert("Icinga CA", key, "Icinga CA", key, [](ASN1_TIME* notBefore, ASN1_TIME* notAfter) {
-		BOOST_REQUIRE(X509_gmtime_adj(notBefore, 0));
-		BOOST_REQUIRE(X509_gmtime_adj(notAfter, LEAF_VALID_FOR + 60 * 60));
-	}).get()));
+	std::array<char, 256> errBuf;
+	ERR_error_string_n(ERR_get_error(), errBuf.data(), errBuf.size());
+	return {errBuf.data()};
 }
 
-BOOST_AUTO_TEST_CASE(iscauptodate_expiring)
+static std::shared_ptr<EVP_PKEY> GetEVP_PKEY(const String& keyfile)
```

Comment:
A more meaningful name would probably helpful (and/or a doc comment). Doesn't this do something like `RsaPrivateKeyFromFile()`?